torchvision/transforms/functional_tensor.py

```diff
@@ -236,3 +276,51 @@ def _blend(img1, img2, ratio):
     # type: (Tensor, Tensor, float) -> Tensor
     bound = 1 if img1.dtype in [torch.half, torch.float32, torch.float64] else 255
     return (ratio * img1 + (1 - ratio) * img2).clamp(0, bound).to(img1.dtype)
+
+
+def _rgb2hsv(img):
+    r, g, b = img[0], img[1], img[2]
+
+    maxc, _ = torch.max(img, dim=0)
+    minc, _ = torch.min(img, dim=0)
+    uv = maxc
+
+    cr = maxc.to(dtype=torch.float32) - minc.to(dtype=torch.float32)
+    s = cr / maxc.to(dtype=torch.float32)
+    rc = (maxc-r).to(dtype=torch.float32) / cr
+    gc = (maxc-g).to(dtype=torch.float32) / cr
+    bc = (maxc-b).to(dtype=torch.float32) / cr
+
+    s = (maxc != minc) * s
+    hr = (maxc == r) * (bc - gc)
+    hg = ((maxc == g) & (maxc != r)) * (2.0 + rc - bc)
+    hb = ((maxc != g) & (maxc != r)) * (4.0 + gc - rc)
+    h = (hr + hg + hb)
+
+    #  torch.fmod and  math.fmod have different precision.
+    h = torch.fmod((h / 6.0 + 1.0), 1.0)
+
+    if img.dtype == torch.uint8:
+        uh = torch.clamp((h*255.0).to(dtype=torch.int32), 0, 255).to(dtype=torch.uint8)
+        us = torch.clamp((s*255.0).to(dtype=torch.int32), 0, 255).to(dtype=torch.uint8)
+    else:
+        uh = torch.clamp(h * 255.0, 0.0, 255.0)
+        us = torch.clamp(s * 255.0, 0.0, 255.0)
+
+    return torch.stack((uh, us, uv))
+
+def _hsv2rgb(img):
+    h, s, v = img[0], img[1], img[2]
+    i = torch.floor(h.to(dtype=torch.float32) * 6.0 / 255.0).to(dtype=torch.int32)
+    f = (h.to(dtype=torch.float32) * 6.0 / 255.0) - i.to(dtype=torch.float32)
+    fs = s.to(dtype=torch.float32) / 255.0
```

Comment:
Similarly here, I think it might be preferable to assume that the inputs are float in 0-1 and in the exact same format as what the `rgb2hsv` returns, so we can avoid performing those divisions in the function.

Reply:
After rgb2hsv I am adding the hue factor which needs to be done in uint8 to avoid round of issues (We can mod there anyway i think). But again should we keep it similar to PIL implementation? I have now simplified the casts so that we dont perform cast on h, s, v, more than once.